openmc/cell.py

```diff
@@ -485,20 +485,23 @@ def get_all_materials(self):
 
         Returns
         -------
-        materials : dict
+        materials : collections.OrderedDict
             Dictionary whose keys are material IDs and values are
             :class:`Material` instances
 
         """
-
         materials = OrderedDict()
         if self.fill_type == 'material':
             materials[self.fill.id] = self.fill
-
-        # Append all Cells in each Cell in the Universe to the dictionary
-        cells = self.get_all_cells()
-        for cell in cells.values():
-            materials.update(cell.get_all_materials())
+        elif self.fill_type == 'distribmat':
+            for m in self.fill:
+                if m is not None:
```

Comment:
Yes, in theory, one of the distributed instances might be a void material, in which case `m is None`